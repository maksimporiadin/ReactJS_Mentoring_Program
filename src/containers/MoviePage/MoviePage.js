import React, { Component } from 'react';

import { Header, MainLayout, Movies, InformPanel, MovieDetails } from '../../components';
import { isEmpty } from '../../shared/utilityMethods';

class MoviePage extends Component {
    state = {
        isShowSearchButton: true,
    }

    componentDidMount() {
        this.uploadMovie();
    }

    componentDidUpdate(prevProps, prevState) {
        if( prevProps.match.params.movieId !== this.props.match.params.movieId ) {
            this.uploadMovie();
        }
    }

    uploadMovie() {
        const id = this.props.match.params.movieId;
        this.props.doMovieInitAction(id);
    }

    render() {
        return (
            <MainLayout>
                <Header isShowSearchButton={ this.state.isShowSearchButton }>
                    <MovieDetails movie={ this.props.movie } loading={ this.props.isLoading }/>
                </Header>
                < InformPanel>
                    {
                        !isEmpty(this.props.movie) &&
                        <div>{`Films by ${this.props.movie.genres.join(' & ')} genre`}</div>
                    }
                </InformPanel>
                    <Movies movies={ this.props.movies } />
            </MainLayout>
        );
    }

}

export default MoviePage;

