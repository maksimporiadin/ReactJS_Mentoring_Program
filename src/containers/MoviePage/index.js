import MoviePage from './MoviePage';
import WithErrorHandler from '../../hoc/WithErrorHandler/WithErrorHandler';
import { doMovieInitAction } from '../../store/actions';
import { connect } from 'react-redux';
import { compose } from 'redux'

const mapStateToProps = state => ({
    isLoading: state.movie.isLoading,
    movie: state.movie.movie,
    movies: state.movies.movies,
});

const withConnect = connect(mapStateToProps, { doMovieInitAction })

export default compose(
  WithErrorHandler,
  withConnect
)(MoviePage)



