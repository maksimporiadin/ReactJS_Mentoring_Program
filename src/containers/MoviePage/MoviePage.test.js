import 'jsdom-global/register';
import React from 'react';
import MoviePage from './MoviePage';
import { shallow } from 'enzyme';

describe('MoviePage', () => {
    let wrapper;
    beforeEach(()=>{
        jest.resetAllMocks();
        wrapper = shallow(<MoviePage />)
        jest.useFakeTimers();
    });

    it('is renders', () => {
        jest.runAllTimers();
        expect(wrapper).toMatchSnapshot();
    });
});