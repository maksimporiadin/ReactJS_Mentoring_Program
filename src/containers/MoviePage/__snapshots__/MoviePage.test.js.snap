// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviePage is renders 1`] = `
<MainLayout>
  <Header
    isShowSearchButton={true}
  />
  <InformPanel />
  <NoFilmsFound />
</MainLayout>
`;
